src/main/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrderQueryFilter.java

```diff
@@ -11,36 +17,55 @@
 @Getter
 public class CasePaymentOrderQueryFilter {
 
-    public static final String CASES_ID = "caseId";
-    public static final String ORDER_REFERENCE = "orderReference";
 
-    private Integer pageSize;
-    private Integer pageNumber;
-    private List<String> listOfIds;
-    private List<String> listOfCasesIds;
+    private List<String> cpoIds;
+    private List<String> caseIds;
+    private Pageable pageable;
 
 
-    public boolean isACasesIdQuery() {
-        return !listOfCasesIds.isEmpty();
+    public boolean isFindByCaseIdQuery() {
+        return !caseIds.isEmpty();
     }
 
     public boolean isAnIdsQuery() {
-        return !listOfIds.isEmpty();
+        return !cpoIds.isEmpty();
     }
 
-    public boolean isAnIdsAndCasesIdQuery() {
-        return isACasesIdQuery() && isAnIdsQuery();
+    public boolean isFilterByBothIdAndCaseId() {
+        return isFindByCaseIdQuery() && isAnIdsQuery();
     }
 
-    public boolean isItAnEmptyCriteria() {
-        return listOfIds.isEmpty() && listOfCasesIds.isEmpty();
+    public boolean noFilters() {
+        return cpoIds.isEmpty() && caseIds.isEmpty();
     }
 
     public List<UUID> getListUUID() {
-        return listOfIds.stream().map(UUID::fromString).collect(Collectors.toList());
+        return cpoIds.stream().map(UUID::fromString).collect(Collectors.toList());
     }
 
     public List<Long> getListOfLongCasesIds() {
-        return listOfCasesIds.stream().map(Long::parseLong).collect(Collectors.toList());
+        return caseIds.stream().map(Long::parseLong).collect(Collectors.toList());
     }
+
+    public Pageable getPageRequest() {
+        return PageRequest.of(
+            getPageNumberMinusOne(),
+            pageable.getPageSize(),
+            Sort.by(
+                Sort.Order.asc(CasePaymentOrderEntity.CASE_ID),
+                Sort.Order.asc(CasePaymentOrderEntity.ORDER_REFERENCE)
+            )
+        );
+    }
+
+    private int getPageNumberMinusOne() {
```

Comment:
```suggestion
    private int getPageNumberZeroBased() {
```
and please add this comment:
//Spring page numbering is zero based

Reply:
OK